[Bug fix] Traces UI: Support filtering on assessments with multiple values (e.g. error and boolean)

dbczumar opened the pull request:
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/dbczumar/mlflow/pull/19262?quickstart=1)

#### Install mlflow from this PR

```
# mlflow
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/19262/merge
# mlflow-skinny
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/19262/merge#subdirectory=libs/skinny
```

For Databricks, use the following command:

```
%sh curl -LsSf https://raw.githubusercontent.com/mlflow/mlflow/HEAD/dev/install-skinny.sh | sh -s pull/19262/merge
```

</p>
</details>

<!-- Remove unused checkboxes except for the patch release section at the bottom -->

### Related Issues/PRs

<!-- Uncomment 'Resolve' if this PR can close the linked items. -->
<!-- Resolve --> #xxx

### What changes are proposed in this pull request?

Traces UI: Support filtering on assessments with multiple values (e.g. error and boolean)

### How is this PR tested?

- [ ] Existing unit/integration tests
- [X] New unit/integration tests
- [X] Manual tests

**Before**

https://github.com/user-attachments/assets/fba72a01-71cb-475e-bb81-f400c027ded4

**After**

https://github.com/user-attachments/assets/ac62dd8c-f01b-4277-8065-bb9b0ff5aca8



### Does this PR require documentation update?

- [X] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

### Release Notes

Fixed a bug that prevented assessments from being used in search filters if they contained errors

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/prompts`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/projects`: MLproject format, project running backends
- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages

<!--
Insert an empty named anchor here to allow jumping to this section with a fragment URL
(e.g. https://github.com/mlflow/mlflow/pull/123#user-content-release-note-category).
Note that GitHub prefixes anchor names in markdown with "user-content-".
-->

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

<!-- Do not modify or remove any text inside the parentheses. Keep both checkboxes below. -->

- [X] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

dbczumar committed:
Add debug logging to AggregationUtils dtype detection

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
rev

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

daniellok-db reviewed (approved):
lg, but looks like some integration test needs to be fixed

dbczumar committed:
progress

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
Add missing i18n key for Error assessment label

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
Add Error option to assessment filter dropdown when errors exist

The filter dropdown for assessments now shows the Error option when
assessmentInfo.containsErrors is true, similar to how bar charts
already handle it.

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
Fix Error label rendering in assessment filter dropdown

getAssessmentValueLabel now handles error value at the start of the
function, before checking dtype. Previously, for boolean dtype
assessments, ERROR_KEY would fall through to the else branch and
render as "null" instead of "Error".

Uses a local constant to avoid circular dependency with AggregationUtils.

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
Add test coverage for getAssessmentValueLabel error handling

Tests verify that ERROR_VALUE ('Error') is handled correctly regardless
of the assessment dtype (boolean, pass-fail, etc).

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>
